src/GraphQL/Types/Scalars/GuidGraphType.cs

```diff
@@ -9,23 +9,45 @@ namespace GraphQL.Types
     public class GuidGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override bool CanParseLiteral(IValue value) => value is StringValue s && Guid.TryParse(s.Value, out _);
-
-        /// <inheritdoc/>
-        public override bool CanParseValue(object value) => value is Guid || value is string s && Guid.TryParse(s, out _);
+        public override object ParseLiteral(IValue value) => value switch
+        {
+            StringValue s => Guid.Parse(s.Value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value) => value is StringValue s ? Guid.Parse(s.Value) : (Guid?)null;
+        public override bool CanParseLiteral(IValue value) => value switch
+        {
+            StringValue s => Guid.TryParse(s.Value, out _),
+            NullValue _ => true,
+            _ => false
+        };
 
         /// <inheritdoc/>
         public override object ParseValue(object value) => value switch
         {
             Guid _ => value, // no boxing
             string s => Guid.Parse(s),
-            _ => null
+            null => null,
+            _ => ThrowValueConversionError(value)
         };
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new StringValue(((Guid)value).ToString());
+        public override bool CanParseValue(object value) => value switch
+        {
+            Guid _ => true,
+            string s => Guid.TryParse(s, out _),
+            null => true,
+            _ => false
+        };
+
+        /// <inheritdoc/>
+        public override object Serialize(object value) => value switch
+        {
+            Guid g => g.ToString(),
```

Comment:
```suggestion
            Guid g => g.ToString("D", CultureInfo.InvariantCulture),
```
Let's be more explicit.